File: authentication_service/src/main/java/huy/project/authentication_service/infrastructure/repository/mapper/UserMapper.java
package huy.project.authentication_service.infrastructure.repository.mapper;

import huy.project.authentication_service.core.domain.dto.request.UpdateUserRequestDto;
import huy.project.authentication_service.core.domain.entity.UserEntity;
import huy.project.authentication_service.infrastructure.repository.model.UserModel;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public abstract class UserMapper {
    public static final UserMapper INSTANCE = Mappers.getMapper(UserMapper.class);

    public abstract UserEntity toEntity(UserModel user);

    public abstract UserModel toModel(UserEntity user);

    public UserEntity toEntity(UserEntity existedUser, UpdateUserRequestDto req) {
        existedUser.setName(req.getName());
        existedUser.setUsername(req.getUsername());
        existedUser.setEmail(req.getEmail());
        return existedUser;
    }
}

File: booking_service/src/core/domain/mapper/user_mapper.go
package mapper

import (
	"booking_service/core/domain/dto/request"
	"booking_service/core/domain/entity"
)

func ToUserEntityUpdate(existedUser *entity.UserEntity, tourist *request.Tourist) {
	existedUser.Email = tourist.Email
	existedUser.LastName = tourist.LastName
	existedUser.FirstName = tourist.FirstName
}

func ToUserEntity(tourist *request.Tourist) *entity.UserEntity {
	return &entity.UserEntity{
		BaseEntity: entity.BaseEntity{
			ID: tourist.TouristID,
		},
		Email:     tourist.Email,
		LastName:  tourist.LastName,
		FirstName: tourist.FirstName,
	}
}
